### PR TITLE
[routing] Add the street name to verbal turn instructions

### PR DESCRIPTION
**Pull request by zyphlar**
Inserts the "nextStreet" variable into the Notifications struct
and then speaks that value with a new "onto" localized word joining
things for TTS output like "in 500 meters, turn left onto Main St."

I omitted the immediate "turn left / then turn right" notifications
since by then it's usually too late to read a street sign and it
can sound repetitious. A TODO is to check highway entrances/exits for
sane language (ideally "Take exit 123 for Main Street, City Center.)

I've added the "onto" localized string and poorly translated it into
all language files. A few languages didn't seem to have equivalents.
I also removed the period from the locale strings so announcements
flow naturally as one sentence.

The best localization would probably be to use formatter strings,
like "in 5000 feet" "turn left onto %s" so that languages with other
orderings or dynamic prepositional syntaxes can sound more natural.

---

Signed DCO.

--- 

I'm not natively a CPP programmer and I don't know how to run the test suite, I'm sure the TTS changes may cause those tests to fail.
Also I'm unsure what the localization/translation process is like, marking the `"onto"` string as desiring of feedback would be ideal.
I've tested this on my Android 12 phone but have no way to test iPhone.

### CODE REVIEW

**Comment by vng** [comment]
Great, thanks!

Please, fix ./data/strings/sounds.txt. Files like ./data/sound-strings/ar.json are generated by ./tools/unix/generate_localizations.sh script.

Will take a look at the code :)

File: routing/turns_tts_text.cpp [hunk 1]
```diff
@@ -62,11 +64,14 @@ string GetTtsText::GetTurnNotification(Notification const & notification) const
...
     return string();
 
-  string const dirStr = GetTextById(GetDirectionTextId(notification));
   if (dirStr.empty())
     return string();
 
   string const distStr = GetTextById(GetDistanceTextId(notification));
+
+  if (notification.m_distanceUnits > 0 && !notification.m_nextStreet.empty())
+    return distStr + " " + dirStr + " " + GetTextById("onto") + " " + notification.m_nextStreet;
```
Comment (vng) [inline routing/turns_tts_text.cpp:73]: We should add "onto" to the sound.txt with all the localizations. And I suspect that it's not an easy task according to languages grammar (not sure that we even can rely on a fixed word's order).

File: routing/turns_notification_manager.cpp [hunk 2]
```diff
@@ -73,12 +73,13 @@ NotificationManager NotificationManager::CreateNotificationManagerForTesting(
 
 string NotificationManager::GenerateTurnText(uint32_t distanceUnits, uint8_t exitNum,
                                              bool useThenInsteadOfDistance, TurnItem const & turn,
-                                             measurement_utils::Units lengthUnits) const
+                                             measurement_utils::Units lengthUnits,
+                                             std::string nextStreet) const
```
Comment (vng) [inline routing/turns_notification_manager.cpp:77]: Better to use std::string const & nextStreet here and below.

**Comment by zyphlar** [comment]
@vng thanks for the review, all this seems fine. I mentioned the possibility of using formatter strings like

```
sprintf("Turn left onto %s", streetName);
```

However this would duplicate half the sounds.txt and as far as I could tell from automated translation suggestions only benefit a few languages, with Japanese and Turkish coming to mind, maybe a couple others. (Most languages seem happy to have the same ordering and I suspect users of these languages may be able to understand even if it sounds artificial... it is a robot after all.)

Your choice, I am happy to do either way. Did you want me to edit strings.txt in this PR while reverting the individual locale files?

**Comment by vng** [comment]
Agree, so:
1st commit: remove useless dots, add "onto" stub into sound.txt
2nd commit: Regenerated strings by generate_localizations.sh
3rd commit: Code modifications

**Comment by zyphlar** [comment]
@vng which do you prefer, copying half of sound.txt like

```
[turn_right]
en: Turn right

[turn_right_onto]
en: Turn right onto %s
```

Or simply adding a single new string and forcing a few languages to have Western-style ordering:

```
[onto]
en: on to
```

**Comment by vng** [comment]
Single one new string [onto]

**Comment by zyphlar** [comment]
No problem will do. How can/should I run the cpp test suite locally to ensure no tests are broken?

**Comment by vng** [comment]
I'm not sure that this particular case is covered, but simply run routing_tests.

**Pull request closed without merging**